api/integrations/lead_tracking/hubspot/services.py

```diff
@@ -12,15 +12,20 @@
 logger = logging.getLogger(__name__)
 
 
-def register_hubspot_tracker(request: Request, user: FFAdminUser | None = None) -> None:
+def register_hubspot_tracker(
+    request: Request,
+    user: FFAdminUser | None = None,
+) -> None:
     hubspot_cookie = request.data.get(HUBSPOT_COOKIE_NAME)
+    utm_data = HubspotTracker.build_utm_data(request.data)
```

Comment:
Looking at this here, it feels like we should be utilising a serializer somewhere (which would likely solve a lot of the duplication of the filtering logic on `ALLOWED_UTM_KEYS`. Which endpoint does this get called from? Can we update its serializer and just pass in the utm_data to this function instead?